src/credentials/attestation_verifier/DeviceAttestationVerifier.h

```diff
@@ -222,6 +220,23 @@ class DeviceAttestationVerifier
         uint16_t productId;
     };
 
+    // Copies the bytes passed to it, and holds the PAI, DAC, and CD for additional verification step
+    class AttestationDeviceInfo
+    {
+    public:
+        AttestationDeviceInfo(const AttestationInfo & attestationInfo);
+        AttestationDeviceInfo(const ByteSpan & attestationElementsBuffer, const ByteSpan paiDerBuffer, const ByteSpan dacDerBuffer);
+
+        ~AttestationDeviceInfo();
+
+        ByteSpan mPaiDerBuffer;       // Buffer containing the PAI certificate from device in DER format.
```

Comment:
`ScopedMemoryBufferWithSize` might be somewhat safer here...

And if we need to add easy conversions from `ScopedMemoryBufferWithSize` to `Span`, we can do that too...

Reply:
Done